x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -885,10 +894,16 @@ public void testGettingDateWithoutCalendar() throws Exception {
         Long randomLongDate = randomNonNegativeLong();
         indexSimpleDocumentWithTrueValues(randomLongDate);
         
-        doWithQuery(SELECT_ALL_FIELDS, (results) -> {
+        // Create simple timezones (in the form of Z/GMT+/-x so that timezones offsets differences between
+        // Java versions (the one server runs on and the one the test runs on) will not skew the tests' results
+        int hoursOffset = randomIntBetween(0, 18);
```

Comment:
Can the timezones vary with more than ±12 hours?

Reply:
@bpintea I took the -18/+18 values from [ZoneOffset](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/time/ZoneOffset.html) itself. These values are the maximum this class supports. Also, at the moment, I think the real range of offsets is [-12, +14].